src/optlang/matrix_interface.py

```diff
@@ -0,0 +1,964 @@
+            k: v for k, v in self.obj_linear_coefs.items() if k in self.variables
+        }
+        self.obj_quadratic_coefs = {
+            k: v
+            for k, v in self.obj_quadratic_coefs.items()
+            if k[0] in self.variables and k[1] in self.variables
+        }
+        self.constraint_lbs = {
+            k: v for k, v in self.constraint_lbs.items() if k in self.constraints
+        }
+        self.constraint_ubs = {
+            k: v for k, v in self.constraint_ubs.items() if k in self.constraints
+        }
+        self.integer_vars = {v for v in self.integer_vars if v in self.variables}
+
+    def rename_constraint(self, old, new):
+        """Rename a constraint.
+
+        Parameters
+        ----------
+        old : str
+            The old name of the constraint.
+        new : str
+            The new name of the constraint.
+
+        Returns
+        -------
+        Nothing.
+
+        """
+        self.reset()
+        self.constraints.remove(old)
+        self.constraints.add(new)
+        name_map = {k: k for k in self.constraints}
+        name_map[old] = new
+        self.constraint_coefs = {
+            (name_map[k[0]], k[1]): v for k, v in self.constraint_coefs.items()
+        }
+        self.constraint_lbs[new] = self.constraint_lbs.pop(old)
+        self.constraint_ubs[new] = self.constraint_ubs.pop(old)
+
+    def rename_variable(self, old, new):
+        """Rename a variable.
+
+        Parameters
+        ----------
+        old : str
+            The old name of the variable.
+        new : str
+            The new name of the variable.
+
+        Returns
+        -------
+        Nothing.
+
+        """
+        self.reset()
+        self.variables.remove(old)
+        self.variables.add(new)
+        name_map = {k: k for k in self.variables}
+        name_map[old] = new
+        self.constraint_coefs = {
+            (k[0], name_map[k[1]]): v for k, v in self.constraint_coefs.items()
+        }
+        self.obj_quadratic_coefs = {
+            (name_map[k[0]], name_map[k[1]]): v
+            for k, v in self.obj_quadratic_coefs.items()
+        }
+        self.obj_linear_coefs = {
+            name_map[k]: v for k, v in self.obj_linear_coefs.items()
+        }
+        self.variable_lbs[new] = self.variable_lbs.pop(old)
+        self.variable_ubs[new] = self.variable_ubs.pop(old)
+
+
+class Variable(interface.Variable):
+    def __init__(self, name, *args, **kwargs):
+        super(Variable, self).__init__(name=name, **kwargs)
+
+    @interface.Variable.type.setter
+    def type(self, value):
+        if self.problem is not None:
+            if value not in _TYPES:
+                raise ValueError(
+                    "This interface cannot handle variables of type '%s'. " % value
+                    + "The following variable types are available: "
+                    + ", ".join(_TYPES)
+                )
+        super(Variable, Variable).type.fset(self, value)
```

Comment:
(I think. Properties are always a bit tricky.)

```suggestion
        super().type.fset(self, value)
```